docs: add privy guide

ankushswar1 opened the pull request.

avasisht23 commented on site/snippets/privy.ts:
```diff
@@ -0,0 +1,30 @@
+// Create a smart account signer from the embedded wallet's viem client
+export const privySigner: SmartAccountSigner = new WalletClientSigner(
+    privyClient
+);
```
nit: newline
```suggestion
);

```

avasisht23 commented on site/snippets/privy.ts:
```diff
@@ -0,0 +1,30 @@
+// Get a viem client from the embedded wallet
+const eip1193provider = await embeddedWallet.getEthereumProvider();
+const privyClient = createWalletClient({
+    account: embeddedWallet?.address,
```
should we recommend the user confirm the wallet exists before creating the client? 
how can it be undefined here?

ankushswar1 replied:
Ah good question - they indeed should confirm that the user has a wallet before continuing. For this particular code snippet, I'm updating it to throw an error if there is no wallet, but I think there may be a better way to structure this altogether - will leave a comment below!

avasisht23 commented on site/smart-accounts/signers/privy.md:
```diff
@@ -0,0 +1,93 @@
+
+# Privy
+
+[Privy](https://privy.io) is the easiest way for web3 developers to onboard users across mobile and desktop, regardless of whether they already have a wallet or not.
```
nit: since it's Alchemy docs, don't want to pick a favorite with language like "easiest" :)
```suggestion
[Privy](https://privy.io) is an easy way for web3 developers to onboard users across mobile and desktop, regardless of whether they already have a wallet or not.
```

avasisht23 reviewed (commented):
lgtm, left comments/nits.

would still like a pass from @moldy530 , but otw will approve by EOD tomorrow after these changes.

ankushswar1 commented:
@avasisht23 Thank you for the comments! They should all be addressed now.

On the topic of "what should the dev do if the user doesn't have an embedded wallet", one note I had mentioned in Slack was:

The other examples all have a pattern where the code snippet is basically broken down into two files:
- the embedded wallet provider’s file (e.g. magic.ts), which exports the SmartAccountSigner
- the alchemy provider’s file

I’ve structured the Privy guide in a similar way for consistency, but it’s a bit less intuitive because Privy is structured as a React SDK, so exporting a signer from one file to another doesn’t really work the same way. 

What do you think about combining the code snippets into a single file? I think that might be more intuitive and would be something a dev could copy-paste into a React component for ex

denniswon commented on site/smart-accounts/signers/privy.md:
```diff
@@ -0,0 +1,93 @@
+    new LightSmartContractAccount({
+      entryPointAddress: "0x...",
+      chain: rpcClient.chain,
+      owner: magicSigner,
```
this should be `privySigner`

ankushswar1 replied:
Ah, silly miss on my end - thank you!

denniswon commented on site/smart-accounts/signers/privy.md:
```diff
@@ -0,0 +1,93 @@
+
+[Privy](https://privy.io) is an easy way for web3 developers to onboard users across mobile and desktop, regardless of whether they already have a wallet or not.
+
+With Privy, you can easily provision **self-custodial embedded wallets** for your users when they login with email, SMS, or social logins, while also enabling web3-native users to use their existing wallets with your app, if they prefer. **It’s one library to onboard all users**, regardless of where they are in their web3 journey.
```
```suggestion
With Privy, you can easily provision **self-custodial embedded wallets** for your users when they login with email, SMS, or social logins, while also enabling web3-native users to use their existing wallets with your app, if they prefer.
```

ankushswar1 replied:
Updated!

denniswon commented on site/smart-accounts/signers/privy.md:
```diff
@@ -0,0 +1,93 @@
+    new LightSmartContractAccount({
+      entryPointAddress: "0x...",
+      chain: rpcClient.chain,
+      owner: magicSigner,
```
```suggestion
      owner: privySigner,
```

denniswon reviewed (commented):
Overall, lgtm

avasisht23 commented:
> I’ve structured the Privy guide in a similar way for consistency, but it’s a bit less intuitive because Privy is structured as a React SDK, so exporting a signer from one file to another doesn’t really work the same way.
> What do you think about combining the code snippets into a single file? I think that might be more intuitive and would be something a dev could copy-paste into a React component for ex

Yeah thanks for flagging. I do like have the code work such that folks can use Privy as a signer regardless of their UI framework (e.g. react vs native, etc.). So, defer to y'all on how you best want to showcase privy and incentivize usage.

re: single file vs. split on docs, fine with single file! choice is yours :)

ankushswar1 commented:
Thanks all for the feedback! I've addressed the comments on the original PR (accidentally using `magicSigner`, updating our description of Privy) and have also combined the code snippets into a single file so it looks a bit more like how it would actually be used in a React app. Let us know if there's anything else!

avasisht23 commented on site/smart-accounts/signers/privy.md:
```diff
@@ -0,0 +1,60 @@
+
+### Create a SmartAccountSigner
+
+First, set up your app with Privy following the [Privy Quickstart](https://docs.privy.io/guide/quickstart). You should also configure your `PrivyProvider` to create embedded wallets for your users when they login, like below:
```
just note that this is specific to react

ankushswar1 replied:
Done! Should I take care of merging, or are you all planning to merge them all at once?

avasisht23 replied:
ill merge!

avasisht23 commented:
hmm @ankushswar1 it looks like this fails now fails our prettier check? can you take a look? 

https://github.com/OMGWINNING/aa-sdk-staging/actions/runs/6433413424/job/17477292865?pr=63

moldy530 commented on site/.vitepress/config.ts:
```diff
@@ -49,6 +49,7 @@ export default defineConfig({
             base: "/smart-accounts/signers",
             link: "/overview",
             items: [
+              { text: "Privy", link: "/privy" },
               { text: "Capsule", link: "/capsule" },
               { text: "Magic.Link", link: "/magic-link" },
               { text: "Turnkey", link: "/turnkey" },
```
nit: keeping things alphabetical

```suggestion
              { text: "Capsule", link: "/capsule" },
              { text: "Magic.Link", link: "/magic-link" },
              { text: "Privy", link: "/privy" },
              { text: "Turnkey", link: "/turnkey" },
```

moldy530 commented:
fixing lints in a separate PR
